quickwit-config/src/index_config.rs

```diff
@@ -19,70 +19,203 @@
 
 use std::ffi::OsStr;
 use std::path::Path;
+use std::time::Duration;
 
 use anyhow::{bail, Context};
 use byte_unit::Byte;
-use quickwit_index_config::FieldMappingEntry;
+use json_comments::StripComments;
+use quickwit_index_config::{FieldMappingEntry, SortBy, SortOrder};
+use quickwit_storage::load_file;
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct DocMapping {
     pub field_mappings: Vec<FieldMappingEntry>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
```

Comment:
This is just a  sggestion, but while we should accept config with no tags, I receommend we keep them when we serialize. This helps with feature discoverability.

```suggestion
```